### PR TITLE
Allow readable size values in cache config

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Allow readable size values in cache config

### CODE REVIEW

File: docs/en/operations/storing-data.md [hunk 1]
```diff
@@ -131,7 +131,7 @@ Example of configuration for versions later or equal to 22.8:
                 <type>cache</type>
                 <disk>s3</disk>
                 <path>/s3_cache/</path>
-                <max_size>10000000</max_size>
```
Comment (antonio2368) [inline docs/en/operations/storing-data.md:134]: you mention 22.8 above but 10Gi won't work for it?

File: tests/queries/0_stateless/02344_describe_cache.reference [hunk 2]
```diff
@@ -1 +1,2 @@
-22548578304	1048576	104857600	1	0	0	0	s3_disk_cache/	0
+22548578304	1048576	104857600	1	0	0	0	s3_cache/	0
+22548578304	1048576	104857600	1	0	0	0	s3_cache_2/	0
```
Comment (antonio2368) [inline tests/queries/0_stateless/02344_describe_cache.reference:2]:
```suggestion
2147483648	1048576	104857600	1	0	0	0	s3_cache/	0
2147483648	1048576	104857600	1	0	0	0	s3_cache_2/	0
```

File: tests/config/config.d/storage_conf.xml [hunk 3]
```diff
@@ -48,16 +48,16 @@
...
+                <max_size>2147483648</max_size>
                 <cache_on_write_operations>1</cache_on_write_operations>
                 <do_not_evict_index_and_mark_files>0</do_not_evict_index_and_mark_files>
             </s3_cache>
             <s3_cache_2>
                 <type>cache</type>
                 <disk>s3_disk_2</disk>
                 <path>s3_cache_2/</path>
-                <max_size>22548578304</max_size>
+                <max_size>2Gi</max_size>
```
Comment (antonio2368) [inline tests/config/config.d/storage_conf.xml:60]: maybe test `max_file_segment_size` including the default?

**Comment by antonio2368** [comment]
I assume we also need to backport it to 22.9

**Comment by kssenii** [comment]
>  assume we also need to backport it to 22.9

Updated the doc that it is available since 22.10, so no need.

**Review by antonio2368 — APPROVED** [review]

**Pull request merged**